File: from-0401-to-0500/0409-longest-palindrome/solution.go
func longestPalindrome(s string) int {
    cnt := make(map[byte]int)
    for i := 0; i < len(s); i ++ {
        cnt[s[i]] += 1
    }

    result := 0
    maxOdd := 0
    for _, val := range cnt {
        if val % 2 == 0 {
            result += val
        } else {
            result += val - 1
            if val > maxOdd {
                maxOdd = 1
            }
        }
    }
    return result + maxOdd
}

File: from-0401-to-0500/0409-longest-palindrome/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        cs = Counter(s)
        
        max_odd = 0
        ans = 0
        for c in cs:
            cnum = cs[c]
            if cnum % 2 == 0:
                ans += cnum
            else:
                ans += (cnum - 1)
                if cnum > max_odd:
                    max_odd = 1
                    
        ans += max_odd
        return ans